services/brig/test/integration/API/User/Util.hs

```diff
@@ -188,15 +190,23 @@ initiateEmailUpdateNoSend brig email uid =
   let emailUpdate = RequestBodyLBS . encode $ EmailUpdate email
    in put (brig . path "/i/self/email" . contentJson . zUser uid . body emailUpdate)
 
-preparePasswordReset :: Brig -> Email -> UserId -> PlainTextPassword -> (MonadIO m, MonadHttp m) => m CompletePasswordReset
-preparePasswordReset brig email uid newpw = do
+preparePasswordReset ::
+  Brig ->
+  DB.ClientState ->
+  Email ->
+  UserId ->
+  PlainTextPassword ->
+  (MonadIO m, MonadHttp m) => m CompletePasswordReset
+preparePasswordReset brig cs email uid newpw = do
   let qry = queryItem "email" (toByteString' email)
   r <- get $ brig . path "/i/users/password-reset-code" . qry
   let lbs = fromMaybe "" $ responseBody r
   let Just pwcode = PasswordResetCode . Ascii.unsafeFromText <$> (lbs ^? key "code" . _String)
-  ident <- PasswordResetIdentityKey <$> mkPasswordResetKey uid
+  ident <- PasswordResetIdentityKey <$> sem (mkPasswordResetKey uid)
   let complete = CompletePasswordReset ident pwcode newpw
   return complete
+  where
+    sem = liftIO . runFinal @IO . interpretClientToIO cs . codeStoreToCassandra @DB.Client
```

Comment:
Can you use `runAppT` here instead?

Reply:
I cannot, right? `runAppT` expects a `PasswordReset` effect at the beginning of the effect row, yet this use site is only about the lower-level code store effect.

Reply:
Fair enough